### ISSUE TITLE
Improve accessibility of form.choice helper

### ISSUE THREAD

**Issue by Stanton** [jadu/pulsar#824]
The pattern doesn't permit the `id` to be used on an element so the `for` attribute points to a non-existent element.

![screen shot 2018-05-08 at 10 52 54](https://user-images.githubusercontent.com/18653/39750764-053ebd0a-52ae-11e8-9daf-c73c64710f04.png)

Pattern should be updated to use a fieldset and legend similar to the recent changes to form.compound.

**Comment by Stanton** [comment]
This should build on #827 to avoid conflicts
